core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -170,6 +182,143 @@ export class ContributionAndReviewService {
+          null,
+          0,
+          AppConstants.SUGGESTIONS_SORT_KEY_DATE,
+          explorationId
+        ).then((fetchSuggestionsResponse) => {
+          const explorationBackendDict = this.explorationObjectFactory.
+          createBackendDictFromExplorationBackendResponse(
+            explorationBackendResponse);
+
+          const exploration: Exploration = this.explorationObjectFactory.
+            createFromBackendDict(explorationBackendDict);
+          const sortedTranslationCards = this.sortTranslationSuggestionsByState(
+            fetchSuggestionsResponse.suggestions,
+            exploration.getStates(),
+            explorationBackendDict.init_state_name
+          );
+          // eslint-disable-next-line
+          const responseSuggestionIdToDetails: {[key: string]: any} = {};
+          const targetIdToDetails = fetchSuggestionsResponse.target_id_to_opportunity_dict;
+          sortedTranslationCards.forEach((suggestion) => {
+          const suggestionDetails = {
+            suggestion: suggestion,
+            details: targetIdToDetails[suggestion.target_id]
+          };
+          responseSuggestionIdToDetails[suggestion.suggestion_id] = suggestionDetails;
+          });
+          return {
+            suggestionIdToDetails: responseSuggestionIdToDetails,
+            more: false
+          };
+        })
+    );
+  }
+
+  // Function to sort translation cards by state.
+  sortTranslationSuggestionsByState(
+      // eslint-disable-next-line
+      translationSuggestions: any[],
+      states: States,
+      initStateName: string | null
+      // eslint-disable-next-line
+  ): any[] {
+    // Obtain the state names in the order of content flow in the lesson.
+    if (!initStateName) {
+      return translationSuggestions;
+    }
+
+    const stateNamesInOrder = this.computeGraphService.
+      computeBfsTraversalOfStates(
+        initStateName,
+        states,
+        initStateName
+      );
+    //  Create an empty map to store translation cards for each state.
+    const translationSuggestionsByState = (
+      // eslint-disable-next-line
+      new Map<string, any[]>());
+
+    // Assign translation cards to the corresponding state in the map.
+    for (const translationSuggestion of translationSuggestions) {
+      const stateName = translationSuggestion.change.state_name;
+      const suggestionsForState = (
+        translationSuggestionsByState.get(stateName) || []);
+      suggestionsForState.push(translationSuggestion);
+      translationSuggestionsByState.set(stateName, suggestionsForState);
+    }
+
+    // Sort translation cards within each state based on the specified criteria
+    // in the order object.
+    for (const stateName of stateNamesInOrder) {
+      const cardsForState = translationSuggestionsByState.get(stateName) || [];
+
+      cardsForState.sort((cardA, cardB) => {
+        return this.compareTranslationSuggestions(cardA, cardB);
+      });
```

Comment:
Can't you just pass in the compare function name, e.g. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#creating_displaying_and_sorting_an_array?

Reply:
Done